v3-nginx/conf.d/default.conf

```diff
@@ -5,22 +5,44 @@ map $http_upgrade $connection_upgrade {
     '' close;
 }
 
+server {
+    listen  ${PORT} default_server;
+
+    location / {
+        return 301 https://$host$request_uri;
+    }
+}
+
 server {
 
     listen ${SSL_PORT} ssl;
-    listen ${PORT};
 
     resolver ${DNS_SERVER} valid=5s;
     server_name ${SERVER_NAME};
     set $upstream ${BACKEND};
 
     ssl_certificate ${PROXY_SSL_CERT};
     ssl_certificate_key ${PROXY_SSL_CERT_KEY};
-    ssl_ciphers ECDH+AESGCM:DH+AESGCM:ECDH+AES256:DH+AES256:ECDH+AES128:DH+AES:ECDH+3DES:DH+3DES:RSA+AESGCM:RSA+AES:RSA+3DES:!aNULL:!MD5:!DSS;
-    ssl_prefer_server_ciphers on;
-    ssl_protocols TLSv1 TLSv1.1 TLSv1.2;
+    ssl_session_timeout 1d;
+    ssl_session_cache shared:MozSSL:10m;
+    ssl_session_tickets off;
+
+    ssl_dhparam /etc/ssl/certs/dhparam-${PROXY_SSL_DH_BITS}.pem;
+
+    ssl_protocols ${PROXY_SSL_PROTOCOLS};
+    ssl_ciphers ${PROXY_SSL_CIPHERS};
+    ssl_prefer_server_ciphers ${PROXY_SSL_PREFER_CIPHERS};
+
+    ssl_stapling ${PROXY_SSL_OCSP_STAPLING};
+    ssl_stapling_verify ${PROXY_SSL_OCSP_STAPLING};
+
     ssl_verify_client ${PROXY_SSL_VERIFY};
 
+    add_header Strict-Transport-Security "max-age=63072000" always;
```

Comment:
What do you think on doing the same here, just moving it to variables, and setting them with sane defaults? I think these values might be good for most cases, but some might need to change them slightly...

Reply:
Thanks @fzipi you're right these do need to configurable but I'm thinking folks will probably want to remove the headers all together at times and I can't think of a way to toggle those on and off with variables.  I am thinking about just pulling those out for now just to get the SSL stuff merged in and maybe revisit the headers.  I also considered just having something like a ADDITIONAL_SERVER_CONFIG as an env variable that would default to empty but allow adding some arbitrary config blocks (i.e. add headers) inside the nginx server directive.

Reply:
🤔 Well, one option could be that, but it might be prone to errors.

Another option is to use the docker-entrypoint.d as upstream does.

I guess for now let's remove the headers part so we can merge this one, and we can discuss headers in a new issue, shall we?